core/variant_op.cpp

```diff
@@ -408,6 +408,31 @@ bool Variant::booleanize() const {
 		_RETURN(sum);                                                                                      \
 	}
 
+// Returns an array of the new axis mappings. e.g. if `p_swizzle == "yz"`,
+// then this method should return `{ Vector3::AXIS_Y, Vector3::AXIS_Z }`.
+// If `p_swizzle` is invalid, e.g. "x5yy", "ab", etc. then it should return an empty Vector.
+static Vector<Vector3::Axis> get_swizzle(String p_swizzle) {
```

Comment:
Is this API efficient? The usage of `String` seems like a red flag. I don't think it makes sense to add swizzling if it's just going to be less efficient than if users manually do it, even if it saves some typing.

Reply:
It'd be more ugly, but I could go back to using predefined `StringNames`. This method is only used to access swizzle combinations, not `x`, `y`, and `z`, so it shouldn't affect the performance of those.
>  I don't think it makes sense to add swizzling if it's just going to be less efficient than if users manually do it, even if it saves some typing.

I'm not sure I agree because:

1. There's a reason why most people write their game logic in GDScript, not C++. Just because GDScript is much less computationally efficient doesn't outweigh it's benefits of simplicity and ease of development.
2. I'm not even convinced this function is that slow/that much slower, it just runs a very simple `for` loop over 2 characters.

That said, once I get this PR working, some benchmarks would not be a bad idea 🙂

Reply:
Alright, after some initial testing, here's what I found. With this script:
```gdscript
var start := 0.0
var vec3 := Vector3.ZERO
var count := 1_000_000


func _process(delta: float) -> void:
	start = OS.get_ticks_usec()
	for i in count:
		vec3.xz = Vector2(10, 20)
	print("Swizzling: ", OS.get_ticks_usec() - start)
	start = OS.get_ticks_usec()
	for i in count:
		vec3 = Vector3(10, vec3.y, 20)
	print("Traditional: ", OS.get_ticks_usec() - start)
```
On a debug build I'm getting these results:
```
--- Debugging process started ---
Swizzling: 1589528
Traditional: 690456
Swizzling: 1599150
Traditional: 694411
Swizzling: 1588243
Traditional: 664318
Swizzling: 1587492
Traditional: 672065
Swizzling: 1593574
Traditional: 703660
Swizzling: 1578223
Traditional: 693721
--- Debugging process stopped ---
```
With the release build from the CI:
```
--- Debugging process started ---
Swizzling: 396098
Traditional: 206678
Swizzling: 411283
Traditional: 203445
Swizzling: 398400
Traditional: 268282
Swizzling: 411427
Traditional: 218208
Swizzling: 411858
Traditional: 205742
Swizzling: 422554
Traditional: 194776
Swizzling: 407783
Traditional: 217953
Swizzling: 394519
Traditional: 183482
Swizzling: 414037
Traditional: 196184
Swizzling: 406828
Traditional: 182722
Swizzling: 684930
Traditional: 344172
Swizzling: 584350
Traditional: 242909
Swizzling: 435674
Traditional: 214129
Swizzling: 408682
Traditional: 192724
Swizzling: 405443
Traditional: 189700
--- Debugging process stopped ---
```

Reply:
@aaronfranke You were right 🙂
Here are the numbers with the new implementation:
Debug build:
```
--- Debugging process started ---
Swizzling: 370295
Traditional: 664358
Swizzling: 336663
Traditional: 744999
Swizzling: 368413
Traditional: 677686
Swizzling: 356991
Traditional: 669693
Swizzling: 358364
Traditional: 661163
Swizzling: 356728
Traditional: 715806
Swizzling: 360910
Traditional: 696443
Swizzling: 354645
Traditional: 688064
Swizzling: 355396
Traditional: 657055
Swizzling: 349508
Traditional: 684757
--- Debugging process stopped ---
```
Release build from CI:
```
--- Debugging process started ---
Swizzling: 119003
Traditional: 188635
Swizzling: 94766
Traditional: 188001
Swizzling: 96210
Traditional: 186763
Swizzling: 94466
Traditional: 184295
Swizzling: 94675
Traditional: 185899
Swizzling: 95879
Traditional: 189787
Swizzling: 90189
Traditional: 214266
Swizzling: 90985
Traditional: 190028
Swizzling: 88276
Traditional: 192563
Swizzling: 98733
Traditional: 185194
Swizzling: 93982
Traditional: 190997
Swizzling: 93655
Traditional: 186486
Swizzling: 99044
Traditional: 182502
Swizzling: 96332
Traditional: 210317
Swizzling: 99491
Traditional: 192851
Swizzling: 95322
Traditional: 183178
Swizzling: 100088
Traditional: 183179
Swizzling: 88661
Traditional: 211196
Swizzling: 95551
Traditional: 193879
Swizzling: 95013
Traditional: 186402
Swizzling: 98606
Traditional: 183122
Swizzling: 95254
Traditional: 187774
Swizzling: 97536
Traditional: 186406
Swizzling: 109574
Traditional: 187458
Swizzling: 102476
Traditional: 181619
Swizzling: 100958
Traditional: 184938
Swizzling: 96302
Traditional: 206708
Swizzling: 101443
Traditional: 182806
Swizzling: 95376
Traditional: 194177
Swizzling: 91510
Traditional: 188593
Swizzling: 88534
Traditional: 189548
Swizzling: 101451
Traditional: 185492
--- Debugging process stopped ---
```
So now, swizzling is about twice as fast as the traditional method.